android/src/main/java/com/swmansion/reanimated/layoutReanimation/SharedTransitionManager.java

```diff
@@ -0,0 +1,400 @@
+  private final Map<Integer, Integer> mSharedTransitionInParentIndex = new HashMap<>();
+  private boolean mIsSharedTransitionActive;
+  private final Map<Integer, Snapshot> mSnapshotRegistry = new HashMap<>();
+  private final Map<Integer, View> mCurrentSharedTransitionViews = new HashMap<>();
+  private View mTransitionContainer;
+  private final List<View> mRemovedSharedViews = new ArrayList<>();
+
+  public SharedTransitionManager(AnimationsManager animationsManager) {
+    mAnimationsManager = animationsManager;
+  }
+
+  protected void notifyAboutNewView(View view) {
+    mAddedSharedViews.add(view);
+  }
+
+  protected void notifyAboutRemovedView(View view) {
+    mRemovedSharedViews.add(view);
+  }
+
+  @Nullable
+  protected View getTransitioningView(int tag) {
+    return mCurrentSharedTransitionViews.get(tag);
+  }
+
+  protected void viewsDidLayout() {
+    setupSharedTransitionForViews(mAddedSharedViews, true);
+    mAddedSharedViews.clear();
+  }
+
+  protected void onViewsRemoving(int[] tagsToDelete) {
+    if (tagsToDelete != null) {
+      visitTreeForTags(tagsToDelete, false);
+
+      setupSharedTransitionForViews(mRemovedSharedViews, false);
+      for (View removedSharedView : mRemovedSharedViews) {
+        visitTree(removedSharedView.getId(), true);
+      }
+      mRemovedSharedViews.clear();
+
+      visitTreeForTags(tagsToDelete, true);
+    }
+  }
+
+  protected void doSnapshotForTopScreenViews(ViewGroup stack) {
+    int screensCount = stack.getChildCount();
+    if (screensCount > 0) {
+      View firstStackChild = stack.getChildAt(0);
+      if (firstStackChild instanceof ViewGroup) {
+        View screen = ((ViewGroup) firstStackChild).getChildAt(0);
+        visitTreeAndMakeSnapshot(screen);
+      } else {
+        Log.e("[Reanimated]", "Unable to recognize screen on stack.");
+      }
+    }
+  }
+
+  protected void setNativeMethods(NativeMethodsHolder nativeMethods) {
+    mNativeMethodsHolder = nativeMethods;
+  }
+
+  private void setupSharedTransitionForViews(List<View> sharedViews, boolean withNewElements) {
+    if (sharedViews.isEmpty()) {
+      return;
+    }
+    sortViewsByTags(sharedViews);
+    List<SharedElement> sharedElements =
+        getSharedElementForCurrentTransition(sharedViews, withNewElements);
+    if (sharedElements.isEmpty()) {
+      return;
+    }
+    setupTransitionContainer();
+    reparentSharedViewsForCurrentTransition(sharedElements);
+    startSharedTransition(sharedElements);
+  }
+
+  private void sortViewsByTags(List<View> views) {
+    /*
+      All shared views during the transition have the same parent. It is problematic if parent
+      view and their children are in the same transition. To keep the valid order in the z-axis,
+      we need to sort views by tags. Parent tag is lower than children tags.
+    */
+    Collections.sort(views, (v1, v2) -> Integer.compare(v2.getId(), v1.getId()));
+  }
+
+  private List<SharedElement> getSharedElementForCurrentTransition(
```

Comment:
```suggestion
  private List<SharedElement> getSharedElementsForCurrentTransition(
```
or even
```suggestion
  private List<SharedElement> sharedElementsForCurrentTransition(
```

Reply:
Let's stick to the convention that method names start with a verb, so the first option works for me.